### PR TITLE
HOSTEDCP-1721: Enable shared ingress for Azure

### PR DESCRIPTION
**Pull request by enxebre**
This enables data plane -> kubernetes.svc traffic and any external dns traffic to use a common router for all HCs in Azure. The initial goal is to start transitioning the code structure (reconcilers, APIs, helpers...) towards a shared ingress oriented solution using haproxy as presented for simplicity. Main changes are:
- Change existing data plane haproxy to forward request to the management kas SVC IP.
- Add a new haproxy in the data plane that listens on the management kas SVC IP and forwards requests to the remote shared ingress.
- Introduce a new common shared ingress for the all the HC fleet management side and uses the proxy protocol to discriminate traffic originated via data plane kubernetes.svc by the dst API of theKAS SVC in the management cluster.
- Update endpoints routes to use the new shared ingress router and stop deploying the per HC router.

Follow ups: PDBs, Network policies, evaluate alternative or adhoc solutions to haproxy, explore how to remove data plane hops, proxy integration, private/public support...
https://issues.redhat.com/browse/HOSTEDCP-1720

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/main.go [hunk 1]
```diff
@@ -213,41 +205,42 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
...
-		if operatorImage == "" {
-			log.Info("operator image is empty, retrying")
-			return false, nil
-		}
-		return true, nil
-	}); err != nil {
-		return fmt.Errorf("failed to find operator image: %w", err)
-	}
-
+	// lookupOperatorImage := func(userSpecifiedImage string) (string, error) {
```
Comment (enxebre) [inline hypershift-operator/main.go:208]: I'll remove all this commented code.

File: hypershift-operator/controllers/nodepool/remote-kas-svc-haproxy/haproxy.cfg [hunk 2]
```diff
@@ -0,0 +1,29 @@
+global
```
[the remark this answers is not in the document]
Comment (enxebre) [reply hypershift-operator/controllers/nodepool/remote-kas-svc-haproxy/haproxy.cfg:1]: Created https://issues.redhat.com/browse/HOSTEDCP-1734 to track.

**Comment by enxebre** [comment]
>    util.go:1466: Metric not found: "hypershift_cluster_silence_alerts"

/test e2e-azure

**Comment by enxebre** [comment]
util.go:1466: Metric not found: "hypershift_cluster_silence_alerts"

/test e2e-azure

File: hypershift-operator/controllers/nodepool/apiserver-haproxy/haproxy.cfg [hunk 3]
```diff
@@ -26,4 +26,4 @@ backend remote_apiserver
   option httpchk GET /version
   option log-health-checks
   default-server inter 10s fall 3 rise 3
-  server controlplane {{ .ExternalAPIAddress }}:{{ .ExternalAPIPort }}
+  server controlplane {{ .RemoteIP }}:{{ .RemotePort }}
```
Comment (phealy) [inline hypershift-operator/controllers/nodepool/apiserver-haproxy/haproxy.cfg:29]: This looks like it's just pointing at the Service ClusterIP - if you were to learn the actual api server pod IPs via DNS service discovery and a headless service ([this](https://www.securityandit.com/network/haproxy-for-service-discovery-in-kubernetes/) blog post covers it pretty well) you could then have haproxy actually balance traffic across the API server pods more intelligently.

Comment (phealy) [reply hypershift-operator/controllers/nodepool/apiserver-haproxy/haproxy.cfg:29]: For future changes :)

**Pull request merged**